expo/expo
# [DO NOT MERGE] fix(inspector-proxy): filter outdated `Debugger.scriptParsed` events to avoid confusing debuggers

# Why

During HMR sessions, every change from Metro is submitted as a separate script. Unfortunately, React Native sends the `Debugger.scriptParsed` events for _all_ scripts loaded during this HMR session, in newest to oldest order.

The result is that the debugger is "confused" and might set breakpoints in different ways, e.g. setting 1 breakpoint causes multiple to be set. That's because of older `Debugger.scriptParsed` events overwriting the actual executed code after an HMR refresh.

This is an experiment to validate this theory and should improve setting breakpoints in the right locations with both vscode and chrome debuggers.

# How

- Added handler that "swallows" older `Debugger.scriptParsed` events.

# Test Plan

- `$ yarn create expo ./test-debugger -t blank@49`
- `$ cd ./test-debugger`
- `$ yarn expo run:ios`
- Make a few changes to the `App.js`, e.g. add some new lines (at least 3x)
- Open debugger
- See that only 1 `Debugger.scriptParsed` event is sent (as opposed to multiple)
- Breakpoints should bind to the right places, even after opening debugger and making changes

# Checklist
